### PR TITLE
[Tests] ES6: Create Strings using Template Literals

### LINKED ISSUE #34291: ES6: Create Strings using Template Literals
**Describe your problem and how to reproduce it:**
You don't have to use any placeholders ie ${myConst} to pass the challenge. And you can simply copy and paste the commented out array provided in the text editor (after removing the asterisks of course). Shouldn't the test be test be checking for the use of placeholders? I know its not listed in the requirements to pass the challenge but placeholders are one of the reasons to use template literals!

**Add a Link to the page with the problem:**
[ES6: Create Strings using Template Literals](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals)

**Tell us about your browser and operating system:**
* Browser Name: Chrome
* Browser Version: 70.0.3538.102
* Operating System: Windows 10

### PR DESCRIPTION
**Pull request by ezioda004**
Added additional tests, I've tested locally. Would be great if someone else can test it out too.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #34291

### CODE REVIEW

**Review by nikrb — CHANGES_REQUESTED** [review]
@ezioda004 tested locally. It behaves but little snafu I think in the solution section, see below.

~~Also, test:curriculum run passes, but I can't find the challenge, e.g. (`cd curriculum`) `./node_modules/.bin/mocha --delay -f 'template'` doesn't show a test for this challenge - I'm not entirely sure what I'm doing, so might just be me.~~
Curriculum test is fine, mocha fgrep option is case sensitive :blush:

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md [hunk 1]
```diff
@@ -77,6 +78,24 @@ const resultDisplayArray = makeList(result.failure);
...
-// solution required
+const result = {
+  success: ["max-length", "no-amd", "prefer-arrow-functions"],
+  failure: ["no-var", "var-on-top", "linebreak"],
+  skipped: ["id-blacklist", "no-dup-keys"]
+};
+function makeList(arr) {
+  "use strict";
+  
+  const resultDisplayArray = result.failure.map(val => `<li class="text-warning">${val}</li>`);
```
[the remark this answers is not in the document]
Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: Ah good catch, fixed.

**Review by nikrb — DISMISSED** [review]

**Comment by nikrb** [comment]
@ezioda004 successfully tested locally

**Comment by ezioda004** [comment]
@nikrb Updated the PR.
I've explicitly mentioned to use an iterator because before there was no mention of using one.

**Comment by nikrb** [comment]
ok @ezioda004 probably should be tested locally again

**Comment by ezioda004** [comment]
@nikrb Tested locally. Can you test too? Thanks.

**Comment by nikrb** [comment]
@ezioda004  successfully tested locally

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md [hunk 2]
```diff
@@ -32,8 +33,9 @@ tests:
   - text: <code>resultDisplayArray</code> is the desired output.
     testString: assert(makeList(result.failure).every((v, i) => v === `<li class="text-warning">${result.failure[i]}</li>` || v === `<li class='text-warning'>${result.failure[i]}</li>`), '<code>resultDisplayArray</code> is the desired output.');
   - text: Template strings were used
```
Comment (ValeraS) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: IMO, we should rephrase this as well. For example: `You need to use a template string` or `Template string should be used`. And use it in the `testString`.

We should describe what result we want to get.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md [hunk 3]
```diff
@@ -32,8 +33,9 @@ tests:
...
-    testString: getUserInput => assert(getUserInput('index').match(/`.*`/g), 'Template strings were not used');
-
+    testString: getUserInput => assert(getUserInput('index').match(/(`.*\${.*}.*`)/g), 'Template strings were not used');
```
Comment (ValeraS) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: Do we need a global flag?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md (hunk 2 above)
Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: How does this sound? "Template string should be used" in the `text` and in `textString` - "Template string or interpolation wasn't used". Could use suggestion here. 😅

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md (hunk 3 above)
Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: Should work without it, fixed.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md (hunk 2 above)
Comment (ValeraS) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: In the most challenges `text` and `textString` contain the same string.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md (hunk 3 above)
Comment (ValeraS) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: And the same in the next `testString`

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md (hunk 2 above)
Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: So, "Template string should be used" is fine?

Comment (ValeraS) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: yes

Comment (ValeraS) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: and for the next test also.

Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: Fixed.

Comment (nikrb) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: > in most challenges `text` and `testString` are the same

I noticed that, it's wrong/lazy isn't it? `text` should be user storyish requirement and `testString` is only displayed on failure, so it should explain what is wrong? 
Impractical to fix them all, but we could fix the ones we update?

Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: I guess its for the convenience but yes I agree it should be a bit more detailed explaining what is wrong with the code.

Comment (ValeraS) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: If you see an error message: `Template string should be used`, what information is missing to understand what is wrong?

Comment (ValeraS) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: IMHO. A user story should describe what we want from campers and a test should check it accordingly. If you can't understand why you don't pass a test then the story is bad or the test is wrong.

Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: Ok - so we dont want solution like this ``arr.map(item => `<li class="text-warning">`+item+`</li>`);`` to the pass the tests but it does technically use "Template string". Shouldnt we mention that "string interpolation" should also be used?

Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md]: Done.

**Comment by ezioda004** [comment]
@ValeraS Does the PR need any more changes?

**Comment by ValeraS** [comment]
@ezioda004 Excuse me for a delay.

LGTM.

**Comment by scissorsneedfoodtoo** [comment]
@ezioda004, everything LGTM! Thanks for all of your hard work here. Also, thank you @nikrb and @ValeraS for all of your suggestions and checking this out locally.
